server/src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -137,7 +137,7 @@ public AllocateUnassignedDecision decideUnassignedShardAllocation(ShardRouting u
         @Override
         protected void reroute(ShardId shardId, String reason) {
             logger.trace("{} scheduling reroute for {}", shardId, reason);
-            routingService.reroute("async_shard_fetch");
+            routingService.reroute("async_shard_fetch shardId " + shardId + " reason [" + reason + "]");
```

Comment:
why is the trace logging above not sufficient? Even though this might be the shard id that triggers the reroute, the actual reroute might be doing allocations for other shards as well.

Reply:
I made this change because the BWC tests have DEBUG as the default logging level.

Reply:
still unclear to me why we should make this change as the logging is available at TRACE level. If debugging an issue, the log level can be adapted accordingly? Also, as I mentioned, the actual reroute might not influence the routing table for the given shard id, so I'm not sure how much that adds

Reply:
Makes sense. I pushed https://github.com/elastic/elasticsearch/pull/39861/commits/61a3abcc1a12314a6543fe3355d009f64eb35fe5 to increase logging-level to debug. Thanks for the suggestion.